### ISSUE TITLE
使用VLLM加速出现CPU高利用率

### ISSUE THREAD

**Issue by Lycoris-Zhang** [FunAudioLLM/CosyVoice#1581]
使用库内自带的fastapi脚本，CosyVoice2-0.5B添加load_vllm=True后实现服务部署，推理中发现CPU占用率高，瞬时单卡CPU占用率可达到50%。GPU利用率约为70%
传入20个请求，发现均有此类情况产生
CosyVoice/runtime/python/fastapi/server.py
cosyvoice = CosyVoice2(args.model_dir, load_jit=True, load_trt=True, fp16=True)加入load_vllm=True
发起API调用后，CPU利用率异常增大

<img width="1746" height="677" alt="Image" src="https://github.com/user-attachments/assets/c685cec3-e25a-4e51-b84c-f6ce82ca4a6d" />

设备环境：
CPU AMD EPYC 9K84 96-Core Processor
GPU NVIDIA H20

**Comment by Lycoris-Zhang** [comment]
补充环境配置
Package                                  Version
---------------------------------------- -------------
absl-py                                  2.3.1
aiofiles                                 23.2.1
aiohappyeyeballs                         2.6.1
aiohttp                                  3.12.15
aiosignal                                1.4.0
airportsdata                             20250811
annotated-types                          0.7.0
antlr4-python3-runtime                   4.9.3
anyio                                    4.10.0
astor                                    0.8.1
async-timeout                            5.0.1
attrs                                    25.3.0
audioread                                3.0.1
beautifulsoup4                           4.13.5
blake3                                   1.0.5
cachetools                               5.5.2
certifi                                  2025.8.3
cffi                                     1.17.1
charset-normalizer                       3.4.3
click                                    8.2.1
cloudpickle                              3.1.1
coloredlogs                              15.0.1
compressed-tensors                       0.9.4
conformer                                0.3.2
contourpy                                1.3.2
cupy-cuda12x                             13.6.0
cycler                                   0.12.1
Cython                                   3.1.3
decorator                                5.2.1
deepspeed                                0.15.1
Deprecated                               1.2.18
depyf                                    0.18.0
diffusers                                0.29.0
dill                                     0.4.0
diskcache                                5.6.3
distro                                   1.9.0
dnspython                                2.8.0
einops                                   0.8.1
email-validator                          2.3.0
exceptiongroup                           1.3.0
fastapi                                  0.115.6
fastapi-cli                              0.0.5
fastrlock                                0.8.3
ffmpy                                    0.6.1
filelock                                 3.19.1
flatbuffers                              25.2.10
fonttools                                4.59.2
frozenlist                               1.7.0
fsspec                                   2024.12.0
gdown                                    5.1.0
gguf                                     0.17.1
google-auth                              2.40.3
google-auth-oauthlib                     1.0.0
googleapis-common-protos                 1.70.0
gradio                                   5.4.0
gradio_client                            1.4.2
grpcio                                   1.57.0
grpcio-tools                             1.57.0
h11                                      0.16.0
hf-xet                                   1.1.9
hjson                                    3.1.0
httpcore                                 1.0.9
httptools                                0.6.4
httpx                                    0.28.1
huggingface-hub                          0.34.4
humanfriendly                            10.0
hydra-core                               1.3.2
HyperPyYAML                              1.2.2
idna                                     3.10
importlib_metadata                       8.4.0
inflect                                  7.3.1
interegular                              0.3.3
Jinja2                                   3.1.6
jiter                                    0.10.0
joblib                                   1.5.2
jsonschema                               4.25.1
jsonschema-specifications                2025.9.1
kaldifst                                 1.7.17
kiwisolver                               1.4.9
lark                                     1.2.2
lazy_loader                              0.4
librosa                                  0.10.2
lightning                                2.2.4
lightning-utilities                      0.15.2
llguidance                               0.7.30
llvmlite                                 0.44.0
lm-format-enforcer                       0.10.12
Markdown                                 3.9
markdown-it-py                           4.0.0
MarkupSafe                               2.1.5
matplotlib                               3.7.5
mdurl                                    0.1.2
mistral_common                           1.8.4
modelscope                               1.20.0
more-itertools                           10.8.0
mpmath                                   1.3.0
msgpack                                  1.1.1
msgspec                                  0.19.0
multidict                                6.6.4
nest-asyncio                             1.6.0
networkx                                 3.1
ninja                                    1.13.0
numba                                    0.61.2
numpy                                    1.26.4
nvidia-cublas-cu12                       12.6.4.1
nvidia-cuda-cupti-cu12                   12.6.80
nvidia-cuda-nvrtc-cu12                   12.6.77
nvidia-cuda-runtime-cu12                 12.6.77
nvidia-cudnn-cu12                        9.5.1.17
nvidia-cufft-cu12                        11.3.0.4
nvidia-cufile-cu12                       1.11.1.6
nvidia-curand-cu12                       10.3.7.77
nvidia-cusolver-cu12                     11.7.1.2
nvidia-cusparse-cu12                     12.5.4.2
nvidia-cusparselt-cu12                   0.6.3
nvidia-nccl-cu12                         2.26.2
nvidia-nvjitlink-cu12                    12.6.85
nvidia-nvtx-cu12                         12.6.77
oauthlib                                 3.3.1
omegaconf                                2.3.0
onnx                                     1.16.0
onnxruntime-gpu                          1.19.2
openai                                   1.106.1
openai-whisper                           20240930
opencv-python-headless                   4.11.0.86
opentelemetry-api                        1.27.0
opentelemetry-exporter-otlp              1.27.0
opentelemetry-exporter-otlp-proto-common 1.27.0
opentelemetry-exporter-otlp-proto-grpc   1.27.0
opentelemetry-exporter-otlp-proto-http   1.27.0
opentelemetry-proto                      1.27.0
opentelemetry-sdk                        1.27.0
opentelemetry-semantic-conventions       0.48b0
opentelemetry-semantic-conventions-ai    0.4.13
orjson                                   3.11.3
outlines                                 0.1.11
outlines_core                            0.1.26
packaging                                24.2
pandas                                   2.3.2
partial-json-parser                      0.2.1.1.post6
pillow                                   11.3.0
pip                                      25.2
platformdirs                             4.4.0
pooch                                    1.8.2
prometheus_client                        0.22.1
prometheus-fastapi-instrumentator        7.1.0
propcache                                0.3.2
protobuf                                 4.25.0
psutil                                   7.0.0
py-cpuinfo                               9.0.0
pyarrow                                  18.1.0
pyasn1                                   0.6.1
pyasn1_modules                           0.4.2
pycountry                                24.6.1
pycparser                                2.22
pydantic                                 2.11.7
pydantic_core                            2.33.2
pydantic-extra-types                     2.10.5
pydub                                    0.25.1
Pygments                                 2.19.2
pyparsing                                3.2.3
PySocks                                  1.7.1
python-dateutil                          2.9.0.post0
python-dotenv                            1.1.1
python-json-logger                       3.3.0
python-multipart                         0.0.12
pytorch-lightning                        2.5.5
pytz                                     2025.2
pyworld                                  0.3.4
PyYAML                                   6.0.2
pyzmq                                    27.0.2
ray                                      2.49.1
referencing                              0.36.2
regex                                    2025.9.1
requests                                 2.32.5
requests-oauthlib                        2.0.0
rich                                     13.7.1
rpds-py                                  0.27.1
rsa                                      4.9.1
ruamel.yaml                              0.18.15
ruamel.yaml.clib                         0.2.12
ruff                                     0.12.12
safehttpx                                0.1.6
safetensors                              0.6.2
scikit-learn                             1.7.1
scipy                                    1.15.3
semantic-version                         2.10.0
sentencepiece                            0.2.1
setuptools                               78.1.1
shellingham                              1.5.4
six                                      1.17.0
sniffio                                  1.3.1
soundfile                                0.12.1
soupsieve                                2.8
soxr                                     1.0.0
starlette                                0.41.3
sympy                                    1.14.0
tensorboard                              2.14.0
tensorboard-data-server                  0.7.2
tensorrt-cu12                            10.0.1
tensorrt-cu12-bindings                   10.0.1
tensorrt-cu12-libs                       10.0.1
threadpoolctl                            3.6.0
tiktoken                                 0.11.0
tokenizers                               0.21.4
tomlkit                                  0.12.0
torch                                    2.7.0
torchaudio                               2.7.0
torchmetrics                             1.8.2
torchvision                              0.22.0
tqdm                                     4.67.1
transformers                             4.51.3
triton                                   3.3.0
typeguard                                4.4.4
typer                                    0.17.4
typing_extensions                        4.15.0
typing-inspection                        0.4.1
tzdata                                   2025.2
urllib3                                  2.5.0
uvicorn                                  0.30.0
uvloop                                   0.21.0
vllm                                     0.9.0
watchfiles                               1.1.0
websockets                               12.0
Werkzeug                                 3.1.3
wetext                                   0.0.4
wget                                     3.2
wheel                                    0.45.1
wrapt                                    1.17.3
xformers                                 0.0.30
xgrammar                                 0.1.19
yarl                                     1.20.1
zipp                                     3.23.0
